### PR TITLE
Add support new targets

### PR DESCRIPTION
**Pull request by caffeine-mgn**
Also I removed old code for development

### CODE REVIEW

**Comment by ionspin** [comment]
Thanks! I'll give it a look as soon as I can.

File: bignum-serialization-kotlinx/build.gradle.kts [hunk 1]
```diff
@@ -66,74 +69,80 @@ signing {
...
+    tvosX64()
+    watchosArm32()
+    watchosArm64()
+    watchosDeviceArm64()
+    watchosX64()
     watchosSimulatorArm64()
-//    mingwX86() Not supported by kotlinx serialization
     mingwX64()
+    wasmJs {
+        browser()
```
Comment (caffeine-mgn) [inline bignum-serialization-kotlinx/build.gradle.kts:115]: Not sure is right

File: bignum-serialization-kotlinx/build.gradle.kts [hunk 2]
```diff
@@ -66,74 +69,80 @@ signing {
...
-                implementation(kotlin(Deps.Common.testAnnotation))
-                implementation(Deps.Common.coroutines)
-            }
+        commonTest.dependencies {
+            implementation(kotlin(Deps.Common.test))
+            implementation(kotlin(Deps.Common.testAnnotation))
+            implementation(Deps.Common.coroutines)
         }
 
-        if (hostOs == primaryDevelopmentOs) {
```
Comment (caffeine-mgn) [inline bignum-serialization-kotlinx/build.gradle.kts:114]: Not need separate target for development and other targets.

File: bignum-serialization-kotlinx/build.gradle.kts (hunk 1 above)
Comment (ionspin) [reply bignum-serialization-kotlinx/build.gradle.kts:115]: Me neither, I'll leave just the browser and see if anyone complains.

**Comment by ionspin** [comment]
Thanks for the pull request! it required some wrangling, and wasmjs/js tests are unstable on mac and windows, but that's letft to be stabilized when wasm is stable in kotlin.

**Pull request merged**